### PR TITLE
Remove x moz errormessage

### PR DESCRIPTION
**Pull request by chrisdavidmills**
As per https://bugzilla.mozilla.org/show_bug.cgi?id=1513890

I added an extra file to contain compat data for global input element attributes, as we didn't have a specific file just for general input element attributes. Let me know what you think ;-)

### CODE REVIEW

**Comment by Elchi3** [comment]
There is also https://github.com/mdn/browser-compat-data/pull/3093 which adds a file for input attributes. 
I thought that @wbamberg maybe has thoughts on how we want to arrange the global input attributes in bcd. If not, we can probably work this out together in the next few days or as a new task in the upcoming sprint. There are a few more "global" input attributes left to be done here.

**Comment by chrisdavidmills** [comment]
Sure, I'm happy to wait for a discussion and decision on this. My addition is not exactly a critical piece of support info; just a nice to have really.

My thinking was that we would have a separate file for "global" input attributes (e.g. available on all input types), and then add support info for attributes that are only supported on specific info types into the JSON files for those types.

Of course, this doesn't mean we'd have to add explicit support info for all attributes; just ones that were added later and so don't fit the "Basic support" details.

**Comment by wbamberg** [comment]
So the current approach for this is that described in https://github.com/mdn/browser-compat-data/pull/1698#issuecomment-390289031, and has us treating different types of input elements as different entities, from a BCD perspective.

At the moment this gives us compat tables like:

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/color#Browser_compatibility
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/range#Browser_compatibility
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#Browser_compatibility

The main reasons for this are:

* there are complex interdependencies between that BCD for "type" and the BCD for the other attributes. If we document them all in one table, we have to try to describe these, and it will not be pretty.
* people use the different input types are if they were different elements: they don't come saying "I want to make an input element" they say "I want a radio button". Presenting them with the info they need on that page seems better for usability.

Documenting compat for other attributes, even global ones, at the level of `input` instead of at the level of the individual types means people have to look in two places to get a complete picture of the compat story for a particular input element, and also seems to mean we might need to describe the interrelationships between this attribute and the `type` attribute.

But that's just my opinion :).

**Comment by chrisdavidmills** [comment]
This all makes sense. So what should we do about global input attributes then? It sounds you are saying we'll have to duplicate the attribute support info across all the attributes. Which won't be any fun.

**Comment by wbamberg** [comment]
The linked comment https://github.com/mdn/browser-compat-data/pull/1698#issuecomment-390289031 talks about this:

> The (related) question is whether we should explicitly enumerate all attributes, even ones whose compat story is the same as "Basic support". Or should we only enumerate ones whose compat story is different from the "Basic support" row? (I think probably we should _not_ enumerate all attributes, but I'm not sure about this.)

**Comment by chrisdavidmills** [comment]
Just to correct my wording from before:

> It sounds you are saying we'll have to duplicate the attribute support info across all the attributes

I meant "It sounds you are saying we'll have to duplicate the attribute support info across all the input types"

To answer Will's followup comment:

> So that recommends that we should only list attributes, including global ones, when we have something interesting to say about them, and otherwise treat them we covered by the "Basic support" data for that input type.

OK, so in this case I believe we do have something interesting to say that isn't covered by Basic support. This still doesn't answer my question of where to put the data in such a case.

**Comment by ExE-Boss** [comment]
This should probably be placed at the top level into `html/elements/input.json`.

**Comment by chrisdavidmills** [comment]
That would probably work for me. I'll wait for a second opinion from Will/Florian.

**Comment by Elchi3** [comment]
That works for me, too. The export "html.elements.input.x-moz-errormessage" is exactly what I would expect.
Just wondering if we're fine with the file name html/elements/input/input_attributes.json, or if we want html/elements/input/input.json or so.

**Comment by chrisdavidmills** [comment]
I guess input.json would be better, as this would allow the file to logically contain data points that aren't just covered by "Basic support", whether they concern attribute support or something else. I'm happy to change it.

I guess one more question is whether we should add "Basic support" info at the input level? I agree with Will's statement about people wanting information on radio buttons or checkboxes, not "the input element", but at the same time it feels weird that the input page's compat table doesn't have a Basic support line, whereas pretty much every other one does.

**Comment by ExE-Boss** [comment]
Let’s add the basic support entry from #1698: https://github.com/mdn/browser-compat-data/blob/5aecc40c8b5e0ae83f18a21d1eb771b1474c6578/html/elements/input.json#L5-L50

File: html/elements/input/input_attributes.json [hunk 1]
```diff
@@ -0,0 +1,58 @@
+{
+  "html": {
+    "elements": {
+      "input": {
```
Comment (ExE-Boss) [inline html/elements/input/input_attributes.json]:
```suggestion
      "input": {
        "__compat": {
          "mdn_url": "https://developer.mozilla.org/docs/Web/HTML/Element/input",
          "support": {
            "chrome": {
              "version_added": "1"
            },
            "chrome_android": {
              "version_added": "1"
            },
            "edge": {
              "version_added": true
            },
            "edge_mobile": {
              "version_added": true
            },
            "firefox": {
              "version_added": "1"
            },
            "firefox_android": {
              "version_added": "4"
            },
            "ie": {
              "version_added": true
            },
            "opera": {
              "version_added": true
            },
            "opera_android": {
              "version_added": true
            },
            "safari": {
              "version_added": "1"
            },
            "safari_ios": {
              "version_added": "1"
            },
            "webview_android": {
              "version_added": "1"
            }
          },
          "status": {
            "experimental": false,
            "standard_track": true,
            "deprecated": false
          }
        },
```

**Comment by chrisdavidmills** [comment]
Thanks @ExE-Boss ; I agree with this, and I think we should change the filename. I've done both.

**Pull request merged**